### PR TITLE
Configure more restrictive Scala compiler options

### PR DESCRIPTION
**Pull request by PawelLipski**
Part of #14

### CODE REVIEW

File: openapi/src/main/scala/com/virtuslab/iat/openapi/DefaultInterpreters.scala [hunk 1]
```diff
@@ -37,7 +37,10 @@ trait DefaultInterpreters {
         apiVersion = Some("v1"), // FIXME: should be in model
         kind = Some("Secret"), // FIXME: should be in model
         metadata = Some(subinterpreter.objectMetaInterpreter(obj, ns)),
-        data = Some(obj.data.mapValues(B64Encoded.apply).toMap)
+        data = {
+          import scala.collection.compat._
```
Comment (pawelprazak) [inline openapi/src/main/scala/com/virtuslab/iat/openapi/DefaultInterpreters.scala:41]: what does this import do? I'm not familiar with this one

File: skuber/src/test/scala/com/virtuslab/iat/skubertest/SkuberLabelExpressionsTest.scala [hunk 2]
```diff
@@ -6,6 +6,8 @@ import org.scalatest.flatspec.AnyFlatSpec
 import org.scalatest.matchers.should.Matchers
 import skuber.LabelSelector._
 
+import scala.language.reflectiveCalls
```
Comment (pawelprazak) [inline skuber/src/test/scala/com/virtuslab/iat/skubertest/SkuberLabelExpressionsTest.scala:9]: where is the reflective call?

Comment (PawelLipski) [reply skuber/src/test/scala/com/virtuslab/iat/skubertest/SkuberLabelExpressionsTest.scala:9]: `is`, `in` etc. on anynomous class declared in `com.virtuslab.iat.dsl.ExpressionsOps#stringToExpression`; fixed now (the class in question is no longer anonymous)

File: openapi/src/main/scala/com/virtuslab/iat/openapi/DefaultInterpreters.scala (hunk 1 above)
Comment (PawelLipski) [reply openapi/src/main/scala/com/virtuslab/iat/openapi/DefaultInterpreters.scala:41]: Done, explained in the comments

**Pull request merged**